mmhuman3d/models/architectures/pymafx.py

```diff
@@ -0,0 +1,409 @@
+# isort: skip_file
+from abc import ABCMeta
+
+import torch
+import torch.nn as nn
+
+from mmhuman3d.models.body_models.smplx import SMPLX_ALL
+from mmhuman3d.models.heads.pymafx_head import (
+    IUV_predict_layer,
+    MAF_Extractor,
+    Mesh_Sampler,
+    get_attention_modules,
+)
+from ...core import constants
+from ..backbones.builder import build_backbone
+from ..heads.builder import build_head
+from .base_architecture import BaseArchitecture
+
+GRID_SIZE = 21
+GLOBAL_FEAT_DIM = 2048
+
+
+def get_fusion_modules(module_keys, ma_feat_dim, grid_feat_dim, n_iter,
+                       out_feat_len):
+
+    feat_fusion = nn.ModuleDict()
+    for k in module_keys:
+        feat_fusion[k] = nn.ModuleList()
+        for i in range(n_iter):
+            feat_fusion[k].append(
+                nn.Linear(grid_feat_dim + ma_feat_dim[k], out_feat_len[k]))
+
+    return feat_fusion
+
+
+class PyMAFX(BaseArchitecture, metaclass=ABCMeta):
+    """PyMAFX Architecture.
+
+    Args:
+        backbone (dict | None, optional): Backbone config dict. Default: None.
+        init_cfg (dict or list[dict], optional): Initialization config dict.
+            Default: None
+    """
+
+    def __init__(self,
+                 backbone,
+                 head,
+                 regressor,
+                 attention_config,
+                 joint_regressor_train_extra,
+                 smpl_model_dir,
+                 mesh_model,
+                 bhf_mode,
+                 maf_on,
+                 body_sfeat_dim,
+                 hf_sfeat_dim,
+                 grid_align,
+                 n_iter=3,
+                 grid_feat=False,
+                 aux_supv_on=True,
+                 hf_aux_supv_on=False,
+                 mlp_dim=[256, 128, 64, 5],
+                 hf_mlp_dim=[256, 128, 64, 5],
+                 loss_uv_regression_weight=0.5,
+                 hf_model_cfg=None,
+                 use_iwp_cam=True,
+                 device=torch.device('cuda'),
+                 init_cfg=None):
```

Comment:
Shall we add docstring here? Especially for those parameters whose meaning cannot be seen from the name.